controllers/common/types.go

```diff
@@ -65,31 +65,30 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 			r.Log.Error(err, "failed to recover chaos")
 			return ctrl.Result{Requeue: true}, nil
 		}
-		return ctrl.Result{}, nil
-	}
+	} else {
+		// Start failure action
+		r.Log.Info("Performing Action")
```

Comment:
Adjust the common reconciler, make sure the chaos object would be udated after Performing or Recovering.